hw/ip/prim/util/primgen.py

```diff
@@ -196,6 +251,54 @@ def _parse_parameter_port_list(parameter_port_list):
     return parameters
 
 
+def _parse_module_header(generic_impl_filepath, module_name):
```

Comment:
We should be able to simplify this function a bit and make it more robust by doing two things:

* Generally, avoid checking for versions, check for features. In this particular case, there might be good reasons to additionally check the version if a call fails, to give users a hint that their Verible version is too old, but do that only for the user if the feature check fails (e.g. try to call Verible with the dumping arguments, and see if it returns an error).

* It feels to me that the compatibility checking code should actually go into the `VeribleVerilogSyntax()` class, throwing an exception if Verible is too old (or other required features are missing). Can you do that?

With those two things together, the code in this function should reduce to something like

```python3
try:
    return _parse_module_header_verible(generic_impl_filepath, module_name)
except Exception as e:
    print(f"Unable to parse file XY with Verible, using regex fallback instead: {e.msg}")
    return _parse_module_header_fallback(generic_impl_filepath, module_name)
```

Reply:
Partially done. The only missing thing is reporting that the Verible version is too old - I'm moving that part to upstream's VeribleVerilogSyntax class. The change prototyped on Opentitan copies of the class: https://github.com/antmicro/opentitan/commit/a0c6489aa1717878431aaf3b87bf7a85ce81214d